exercises/etl/src/example.h

```diff
@@ -0,0 +1,16 @@
+#ifndef ETL_H
+#define ETL_H
+
+typedef struct {
+   int value;
+   const char *keys;
+} legacy_map;
+
+typedef struct {
+   char key;
+   int value;
+} new_map;
+
+int convert(legacy_map * input, int input_len, new_map ** output);
```

Comment:
How about adding a `const` to `legacy_map` since it shouldn't be touched?

Reply:
Does it even need to be a pointer? The legacy map is a fixed length array (as far as `convert`is concerned I mean). This could `potentially change to `int convert (legacy_map input[], int input_len, new_map ** output);`. Depends how you feel about having both pointer type and array type parameters. Personally I don't feel it's a major concern except that it follows C's general idea of being as restrictive as possible  for types and makes the types explicit as to which parameter is the input and which is the output.

Reply:
I don't think that I follow regarding it being a fixed length array. Isn't the actual length of the array unknown?

If you're not putting an array size, `legacy_map input[]` as a parameter is AFAIK identical to `legacy_map *input`. It doesn't pass by value and it doesn't even restrict the size of the array (any provided size is ignored). The only real difference with array types and pointer types as arguments is the behavior of `sizeof`. Only if you do something like `input[static n]` do you put any constraint on the passed array. C is bizarre sometimes.